src/guide/deployment/lamp.rst

```diff
@@ -0,0 +1,160 @@
+
+Note that we used ``myservice@'%'`` as user name. This generally allows access from everywhere (``%``). Our firewall rules make sure that MySQL is only reachable via the SRV interface and only from other VMs in the *same* resource group. See :ref:`firewall` for details.
+
+
+Apache / FPM
+************
+
+You configure Apache and FPM in one step. With the Flying Circus ``lamp`` role you define an Apache docroot which is properly hooked up with FPM.
+
+But first, there are some generic Apache settings, which are used for *all* sites. :
+
+.. code-block:: Nix
+
+    # /etc/local/nixos/apache.nix
+    { pkgs, ... }:
+
+    {
+
+      flyingcircus.roles.lamp = {
+
+        # Configure PHP version:
+        php = pkgs.lamp_php74;
+
+        # Additional Apache configuration
+        apache_conf = ''
+          MaxRequestWorkers 5
+        '';
+
+        # php.ini settings:
+        php_ini = ''
+          ; max filesize
+          upload_max_filesize = 200M
+          post_max_size = 200M
+
+
+          date.timezone = Europe/Berlin
+
+          ; Putting sessions in redis is a good idea.
+          ; session.save_handler = redis
+          ; session.save_path = "tcp://myvm00.fcio.net:6379?auth=<secret>"
+        '';
+
+      };
+    }
+
+Now configure a docroot. It is best to create a separate service user for each project and separate the vhost configuration defining listening port and the corresponding docroot:
+
+.. code-block:: Nix
+
+    # /etc/local/nixos/myservice-apache.nix
+    { ... }:
+    {
+      flyingcircus.roles.lamp = {
+        hosts = [
+          { port = 8000;
+            docroot = "/srv/s-myserviceuser/application.git/docroot";
+          }
+        ];
+      };
+    }
+
+
+This setting configures both Apache *and* FPM automatically. To enable the configuration, call ``sudo fc-manage -b``
+
+The detailed documentation for this component is: :ref:`nixos-lamp`.
+
+
+NGINX
+*****
+
+The NGINX component manages SSL termination and routing to Apache.
+
+.. code-block:: nix
+
+    # /etc/local/nixos/myservice-nginx.nix
+    { ... }:
+    {
+      flyingcircus.services.nginx.virtualHosts = {
+        "www.example.com"  = {
+          serverAliases = [ "example.com" ];
+          forceSSL = true;  # Create redirect from http to https
+          enableACME = true;  # Create Let's Encrypt certificate
+          extraConfig = ''
+            # Additional verbatim NGINX config.
+            add_header Strict-Transport-Security max-age=31536000;
+          '';
+          locations = {
+            "/" = {
+              # Pass request to Apache. The port must match the Apache configuration
+              proxyPass = "http://myvm00.fcio.net:8000";
+              extraConfig = ''
+                # Additional verbatim NGINX config.
+              '';
+            };
+          };
+        };
+      };
+    }
+
+You will find more details on NGINX configuration at :ref:`nixos-webgateway`.
+
+
+
```

Comment:
One thing  that would be interesting to show is that you can

a) put all this config into a single file and 
b) use variables to e.g. consistently refer to the virtual host port (8000)